tinygrad/helpers.py

```diff
@@ -19,8 +19,8 @@ def flatten(l:Iterator): return [item for sublist in l for item in sublist]
 def mnum(i) -> str: return str(i) if i >= 0 else f"m{-i}"
 def fromimport(mod, frm): return getattr(__import__(mod, fromlist=[frm]), frm)
 def merge_dicts(ds:Iterable[Dict]) -> Dict:
-  kvs = set([(k,v) for d in ds for k,v in d.items()])
-  assert len(kvs) == len(set(kv[0] for kv in kvs)), f"cannot merge, {kvs} contains different values for the same key"
+  kvs = {(k,v) for d in ds for k,v in d.items()}
+  assert len(kvs) == len({kv[0] for kv in kvs}), f"cannot merge, {kvs} contains different values for the same key"
```

Comment:
Saves creation of another set:
```suggestion
  merged = {k:v for k,v in kvs}
  assert len(kvs) ==  @len(merged), f"cannot merge, {kvs} contains different values for the same key"
  return merged
```

Reply:
I don't mind the assert statement duplication, that one will be skipped with `-O`. Also - `{kv[0] for kv in kvs} =|= {k:v for k,v in kvs}` if I am not wrong.